src/Internal/RejectedPromise.php

```diff
@@ -25,16 +26,31 @@ public function __construct(\Throwable $reason)
         $this->reason = $reason;
     }
 
+    /** @throws void */
     public function __destruct()
     {
         if ($this->handled) {
             return;
         }
 
-        $message = 'Unhandled promise rejection with ' . \get_class($this->reason) . ': ' . $this->reason->getMessage() . ' in ' . $this->reason->getFile() . ':' . $this->reason->getLine() . PHP_EOL;
-        $message .= 'Stack trace:' . PHP_EOL . $this->reason->getTraceAsString();
+        $handler = set_rejection_handler(null);
```

Comment:
Hi, it's one time use because this is inside the promise that is not handled when rejected. It will only ever run once. It might run again for another promise, but not this specific one.

Reply:
But there is no connection between the handler and the promise here. I can't explicitly specify which promise this handler is for. And if the user completes another promise without handling the exception, they will not be notified about it, while the promise for which I would like to define a handler will complete without a handler.

Are you considering the possibility of adding a function to set a global persistent handler for all promises? Or add the ability to specify a chain of promises for the handler?

Reply:
Hey @roxblnfk, took a closer look at this with @clue and it seems like following exceptions are currently not being handled correctly. We'll look into this, create some additional tests for multiple exceptions to define our expected behavior and implement a fix.

Thanks for taking the time and bringing this up :+1:

Reply:
Ok I think I misunderstood your initial comment. You are right, this shouldn't be a one use only handler but a persistent one instead. We're looking into adding additional tests and a fix for this.

Reply:
Thanks.